internal/pkg/skuba/kubernetes/resources_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package kubernetes
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_DoesSecretResourceExistWithError(t *testing.T) {
+	fakeWorker := corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "worker",
+		},
+	}
+
+	fakeMaster := corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "master",
+			Labels: map[string]string{"node-role.kubernetes.io/master": ""},
+		},
+	}
+
+	tests := []struct {
+		name             string
+		fakeClientset    *fake.Clientset
+		searchNamespace  string
+		searchSecret     string
+		expectReturn     bool
+		expectErrMessage error
+		fakeErrMessage   error
+	}{
+		{
+			name: "secret exist with no error",
+			fakeClientset: fake.NewSimpleClientset(
```

Comment:
The same `fakeClientset` exists in three test data, I would suggest to create one on the top then reuse to avoid duplicated creation.